shell/server/pre-meteor.js

```diff
@@ -210,7 +210,24 @@ function checkMagic(buf, magic) {
   return true;
 }
 
-function serveStaticAsset(req, res) {
+function serveSelfTest(req, res) {
+  inMeteor(() => {
+    if (req.method === 'GET' &&
+        req.url === '/') {
+      return serveStaticAsset(req, res, {
```

Comment:
I can live with that. FWIW I overloaded `staticServeAsset` because I don't
love the idea of writing copy-pasta for OPTIONS. I could extract _that_
into a function.